File: src/main/kotlin/com/intellij/plugin/gamification/listeners/GameEventListener.java
package com.intellij.plugin.gamification.listeners;

import com.intellij.util.messages.Topic;

public interface GameEventListener {
    Topic<GameEventListener> TOPIC = new Topic<>(GameEventListener.class);

    /**
     * Called when a level has changed.
     */
    default void levelChanged() { }

    /**
     * Called when a progress has changed.
     */
    default void progressChanged() { }
}

File: src/main/kotlin/com/intellij/plugin/gamification/listeners/GameEventListener.kt
package com.intellij.plugin.gamification.listeners

import java.util.EventListener

interface GameEventListener : EventListener {

    /**
     * Called when a level has changed.
     */
    fun levelChanged(event: GameEvent) {}

    /**
     * Called when a progress has changed.
     */
    fun progressChanged(event: GameEvent) {}
}

data class GameEvent(val level: Int, val progress: Int)
